message Recoded {
  message Metadata {
    optional bytes version = 1;
    optional bytes source_commit = 2;
    optional bytes binary_sha256 = 3;
    optional int64 binary_timestamp = 4;
  };
  optional Metadata metadata = 1;

  message Block {
    optional int64 size = 1;
    optional bytes literal = 2;
    optional bool skip_coded = 3;  // Always true when present.
    optional bytes cabac = 4;
    optional bool length_parity = 5; // To detect presence of x264 padding.
    optional bytes last_byte = 6; // Last octet (zero or x264 signature bits)
  };
  repeated Block block = 2;
};
